ion-schema/src/ion_schema_version.rs

```diff
@@ -44,3 +44,40 @@ pub(crate) trait SinceISL_2_0: IslVersion { }
 impl SinceISL_2_0 for ISL_2_0 {}
 
  */
+
+use std::marker::PhantomData;
+use std::ops::Deref;
+
+/// Wrapper for data that needs to be passed around with an ISL version—for example, builder methods
+/// for some constraints can accept `Versioned<TypeArgument>`s.
+///
+/// Technically, this is a smart pointer that (rather than managing the ownership or memory of the
+/// wrapped value) encodes Ion Schema version information along with the wrapped value.
```

Comment:
🗺️ I'm not completely sure whether this should be an opaque type or a pseudo-smartpointer. I'm open to suggestions. I'm calling it a smart pointer for now because it implements `Deref`, so I also tried to follow the other best practices for functions on smart pointers.